bundles/org.openhab.binding.bmwconnecteddrive/src/main/java/org/openhab/binding/bmwconnecteddrive/internal/handler/ConnectedDriveBridgeHandler.java

```diff
@@ -0,0 +1,196 @@
+import org.openhab.binding.bmwconnecteddrive.internal.utils.Constants;
+import org.openhab.binding.bmwconnecteddrive.internal.utils.Converter;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ConnectedDriveBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class ConnectedDriveBridgeHandler extends BaseBridgeHandler implements StringResponseCallback {
+    private final Logger logger = LoggerFactory.getLogger(ConnectedDriveBridgeHandler.class);
+    private HttpClientFactory httpClientFactory;
+    private Optional<VehicleDiscovery> discoveryService = Optional.empty();
+    private Optional<ConnectedDriveProxy> proxy = Optional.empty();
+    private Optional<ConnectedDriveConfiguration> configuration = Optional.empty();
+    private Optional<ScheduledFuture<?>> initializerJob = Optional.empty();
+    private Optional<String> troubleshootFingerprint = Optional.empty();
+
+    public ConnectedDriveBridgeHandler(Bridge bridge, HttpClientFactory hcf) {
+        super(bridge);
+        httpClientFactory = hcf;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no commands available
+    }
+
+    @Override
+    public void initialize() {
+        troubleshootFingerprint = Optional.empty();
+        updateStatus(ThingStatus.UNKNOWN);
+        configuration = Optional.of(getConfigAs(ConnectedDriveConfiguration.class));
+        if (configuration.isPresent()) {
+            proxy = Optional.of(new ConnectedDriveProxy(httpClientFactory, configuration.get()));
+            // give the system some time to create all predefined Vehicles
+            initializerJob = Optional.of(scheduler.schedule(this::requestVehicles, 5, TimeUnit.SECONDS));
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (initializerJob.isPresent()) {
+            initializerJob.get().cancel(true);
+        }
+    }
+
+    public void requestVehicles() {
+        if (proxy.isPresent()) {
+            proxy.get().requestVehicles(this);
+        }
+    }
+
+    public String getDiscoveryFingerprint() {
+        if (troubleshootFingerprint.isPresent()) {
+            VehiclesContainer container = Converter.getGson().fromJson(troubleshootFingerprint.get(),
+                    VehiclesContainer.class);
```

Comment:
You should catch the parsing exception that is thrown by gson.

Reply:
Done for all occurrences